src/App.jsx

```diff
@@ -14,33 +15,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function sortByAlphabetically(copyAlphabeticallyArray) {
+  const sortAlphabeticallyArray = [...copyAlphabeticallyArray];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  return sortAlphabeticallyArray.sort((good1, good2) =>
+    good1.localeCompare(good2),
+  );
+}
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+function sortByLength(copyLengthArray) {
+  const sortLengthArray = [...copyLengthArray];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return sortLengthArray.sort((good1, good2) => good1.length - good2.length);
+}
+
+function reverseCopyArray(copyReverseArray) {
+  const arrayReverse = [...copyReverseArray];
+
+  return arrayReverse.reverse();
+}
+
+export function App() {
+  const [goodValue, setGoodValue] = useState([...goodsFromServer]);
+  const [flag, setFlag] = useState(false);
+  const [activeMethod, setActiveMethod] = useState(null);
```

Comment:
1. sortType or sortField will be more meaningful name
2. you can use an empty string (`''`) instead of a `null` to not set two possible types to a value
```suggestion
  const [sortType, setSortType] = useState('');
```